src/spark/utils.py

```diff
@@ -44,17 +46,23 @@ def _get_s3_conf() -> Dict[str, str]:
     Helper function to get S3 configuration for MinIO.
     """
     return {
-        "spark.hadoop.fs.s3a.endpoint": not_falsy(
-            os.environ.get("MINIO_URL"), "MINIO_URL"
+        "spark.hadoop.fs.s3a.endpoint": str(
+            not_falsy(os.environ.get("MINIO_URL"), "MINIO_URL")
         ),
-        "spark.hadoop.fs.s3a.access.key": not_falsy(
-            os.environ.get("MINIO_ACCESS_KEY"), "MINIO_ACCESS_KEY"
+        "spark.hadoop.fs.s3a.access.key": str(
+            not_falsy(os.environ.get("MINIO_ACCESS_KEY"), "MINIO_ACCESS_KEY")
         ),
-        "spark.hadoop.fs.s3a.secret.key": not_falsy(
-            os.environ.get("MINIO_SECRET_KEY"), "MINIO_SECRET_KEY"
+        "spark.hadoop.fs.s3a.secret.key": str(
+            not_falsy(os.environ.get("MINIO_SECRET_KEY"), "MINIO_SECRET_KEY")
         ),
```

Comment:
Not sure why you need the str() calls here

Reply:
oh, just to stop my linter complaining.